source/common/router/config_impl.cc

```diff
@@ -143,6 +143,14 @@ RouteEntryImplBase::RouteEntryImplBase(const VirtualHostImpl& vhost, const Json:
           {Http::LowerCaseString(header->getString("key")), header->getString("value")});
     }
   }
+
+  // Only set include_vh_rate_limits_ to true if the rate limit policy for the route is empty
+  // or the route set `include_vh_rate_limits` to true.
+  if (rate_limit_policy_.empty() || route.getBoolean("include_vh_rate_limits", false)) {
```

Comment:
nit: include_vh_rate_limits = rate_limit_policy_.empty() || route.getBoolean("include_vh_rate_limits", false); for brevity